docs/en/04_dataobjects.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Using DataObjects with Fluent
+summary: Configuring DataObjects for use with Fluent, including CMS fields, extensions, and versioned objects.
+icon: database
+---
+
+# Using dataObjects with fluent
+
+## CMS fields
+
+With each of the possible extensions that come with Fluent, it's necessary to configure your DataObjects to
+correctly delegate your `$fields` to extensions. A different approach may be needed for each DataObject, depending
+on how it generates it's fields.
+
+In all cases, it is necessary to ensure that the `updateCMSFields()` extension method is called once (and
+only once) on each `FieldList` object.
+
+### Default field scaffolder
+
+If using a default field scaffolder (such as [`SiteTree::getCMSFields()`](api:SilverStripe\CMS\Model\SiteTree::getCMSFields()) or `DataObject::getCMSFields`)
+it is necessary to know that these methods both, by default, will call the
+`extend('updateCMSFields', $fields)` extension before returning.
+
+If you wish to add any translated form fields to the result of this call, then you should use [`DataObject::beforeUpdateCMSFields()`](api:SilverStripe\ORM\DataObject::beforeUpdateCMSFields()):
+
+```php
+namespace App\Pages;
+
+class MyPage extends SiteTree
+{
+    public function getCMSFields()
+    {
+        // Adding the Description field early will allow FluentField to decorate this with the appropriate
+        // CSS classes.
+        $this->beforeUpdateCMSFields(function ($fields) {
+            $fields->addFieldToTab('Root.Main', new TextField('Description'));
```

Comment:
```suggestion
            $fields->addFieldToTab('Root.Main', TextField::create('Description'));
```
Please also add the `use` statement for `TextField`

Reply:
Updated